crates/biome_js_analyze/src/lint/nursery/no_static_element_interactions.rs

```diff
@@ -0,0 +1,353 @@
+            "onMouseOver", "onMouseUp"
+        ]);
+        m.insert("selection", vec!["onSelect"]);
+        m.insert("touch", vec!["onTouchCancel", "onTouchEnd", "onTouchMove", "onTouchStart"]);
+        m.insert("ui", vec!["onScroll"]);
+        m.insert("wheel", vec!["onWheel"]);
+        m.insert("media", vec![
+            "onAbort", "onCanPlay", "onCanPlayThrough", "onDurationChange", "onEmptied",
+            "onEncrypted", "onEnded", "onError", "onLoadedData", "onLoadedMetadata", "onLoadStart",
+            "onPause", "onPlay", "onPlaying", "onProgress", "onRateChange", "onSeeked", "onSeeking",
+            "onStalled", "onSuspend", "onTimeUpdate", "onVolumeChange", "onWaiting"
+        ]);
+        m.insert("image", vec!["onLoad", "onError"]);
+        m.insert("animation", vec!["onAnimationStart", "onAnimationEnd", "onAnimationIteration"]);
+        m.insert("transition", vec!["onTransitionEnd"]);
+        m
+    };
+}
+
+impl Rule for NoStaticElementInteractions {
+    type Query = Aria<AnyJsxElement>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let aria_roles = ctx.aria_roles();
+        let element_name = node.name().ok()?.as_jsx_name()?.value_token().ok()?;
+        let attributes = extract_attrs(&node.attributes());
+        let elm = element_name.text_trimmed();
+
+        if let Some(attributes_ref) = attributes.as_ref() {
+            if !is_interactive_handler_present(attributes_ref) {
+                return None;
+            }
+        } else {
+            return None;
+        }
+
+        let default_attributes = FxHashMap::default();
+        let attributes_ref = attributes.as_ref().unwrap_or(&default_attributes);
+
+        let attributes_option = if !attributes_ref.is_empty() {
+            Some(attributes_ref.clone())
+        } else {
+            None
+        };
+
+        if has_truthy_ariahidden_attr(node) {
+            return None;
+        }
+
+        let is_valid_element = match elm {
+            "section" => ["aria-label", "aria-labelledby"].iter().any(|&attr_name| {
+                node.find_attribute_by_name(attr_name)
+                    .map_or(false, |attr| {
+                        attr.as_static_value()
+                            .map_or(false, |val| !val.text().is_empty())
+                    })
+            }),
+            "a" => node.find_attribute_by_name("href").map_or(false, |attr| {
+                attr.as_static_value()
+                    .map_or(false, |val| !val.text().is_empty())
+            }),
+            _ => {
+                (!aria_roles.is_not_interactive_element(
+                    element_name.text_trimmed(),
+                    attributes_option.clone(),
+                ) && !is_invalid_element(elm))
+                    || is_valid_element(elm)
```

Comment:
could you leave a brief comment?

Reply:
@unvalley 
- is_invalid_element: 
- is_valid_element

These two methods were created to bridge the differences between the test cases of eslint-plugin-jsx-a11y and Biome's definition of interactive elements. 
However, we will update Biome's is_not_interactive_element() method based on official information from sources like W3C, rather than relying on eslint-plugin-jsx-a11y.

Reply:
I think the if condition is complicated a bit, so comments are helpful to understand what that means.

Reply:
@unvalley 
Sorry, I misunderstood...
fixed  in this commit  https://github.com/biomejs/biome/pull/2981/commits/f4f65cb0c4d4b58d02782f593797308d13c731cb